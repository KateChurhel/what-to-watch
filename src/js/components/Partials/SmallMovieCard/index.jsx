// libraries
import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
// constants
import { URL_IMAGE } from '../../../constants/url';

const SmallMovieCard = ({ poster, cardTitle, link }) => (cardTitle ? (
  <Link className="small-movie-card catalog__movies-card small-movie-card__link" to={link}>
    <div className={`small-movie-card__image ${poster ? '' : 'small-movie-card__image--no-image'}`}>
      <img
        alt={cardTitle}
        height="450"
        loading="lazy"
        src={poster ? `${URL_IMAGE.poster}${poster}` : `${process.env.PUBLIC_URL}/img/no-poster.jpeg`}
        width="300"
      />
    </div>
    <div className="small-movie-card__info">
      <h3 className="small-movie-card__title">{cardTitle}</h3>
    </div>
  </Link>
) : null);

SmallMovieCard.defaultProps = {
  cardTitle: '',
  poster: '',
  link: '',
};

SmallMovieCard.propTypes = {
  poster: PropTypes.string,
  cardTitle: PropTypes.string,
  link: PropTypes.string,
};

export default SmallMovieCard;
