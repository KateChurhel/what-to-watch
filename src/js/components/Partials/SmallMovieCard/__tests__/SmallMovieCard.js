import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import SmallMovieCard from '../index';

const MOCK_DATA = {
  cardTitle: 'Title',
  link: '/link',
};

describe('Small Movie Card: ', () => {
  it('correctly renders with data', () => {
    const tree = shallow(<SmallMovieCard {...MOCK_DATA} poster="/poster.jpg" />);
    expect(toJSON(tree)).toMatchSnapshot();
  });

  it('correctly renders without poster', () => {
    const tree = shallow(<SmallMovieCard {...MOCK_DATA} />);
    expect(toJSON(tree)).toMatchSnapshot();
  });

  it('correctly renders without data', () => {
    const tree = shallow(<SmallMovieCard />);
    expect(toJSON(tree)).toMatchSnapshot();
  });
});
