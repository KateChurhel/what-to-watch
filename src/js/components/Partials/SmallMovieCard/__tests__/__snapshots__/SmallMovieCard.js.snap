// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Small Movie Card:  correctly renders with data 1`] = `
<a
  className="small-movie-card catalog__movies-card small-movie-card__link"
  href="/link"
>
  <div
    className="small-movie-card__image "
  >
    <img
      alt="Title"
      height="450"
      loading="lazy"
      src="https://image.tmdb.org/t/p/w300/poster.jpg"
      width="300"
    />
  </div>
  <div
    className="small-movie-card__info"
  >
    <h3
      className="small-movie-card__title"
    >
      Title
    </h3>
  </div>
</a>
`;

exports[`Small Movie Card:  correctly renders without data 1`] = `""`;

exports[`Small Movie Card:  correctly renders without poster 1`] = `
<a
  className="small-movie-card catalog__movies-card small-movie-card__link"
  href="/link"
>
  <div
    className="small-movie-card__image small-movie-card__image--no-image"
  >
    <img
      alt="Title"
      height="450"
      loading="lazy"
      src="/img/no-poster.jpeg"
      width="300"
    />
  </div>
  <div
    className="small-movie-card__info"
  >
    <h3
      className="small-movie-card__title"
    >
      Title
    </h3>
  </div>
</a>
`;
